FactorioWebInterface/Services/WebAccountManager.cs

```diff
@@ -0,0 +1,97 @@
+﻿using FactorioWebInterface.Data;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace FactorioWebInterface.Services
+{
+    public interface IWebAccountManager
+    {
+        Task<ApplicationUser> FindByNameAsync(string username);
+        Task<ApplicationUser> FindByIdAsync(string id);
+        Task<ApplicationUser> GetUserAsync(ClaimsPrincipal user);
+        Task CreateAccountAsync(string username, string password, string[] roles);
+        Task ChangePasswordAsync(ApplicationUser user, string oldPassword, string newPassword);
+        Task AddRoleAsync(ApplicationUser user, string role);
+        Task RemoveRoleAsync(ApplicationUser user, string role);
+        Task SuspendAccountAsync(ApplicationUser user, bool suspended = true);
+    }
+
+    public class WebAccountManager : IWebAccountManager
+    {
+        private readonly SignInManager<ApplicationUser> _signInManager;
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly ILogger<WebAccountManager> _logger;
+
+        public WebAccountManager(UserManager<ApplicationUser> userManager,
+            SignInManager<ApplicationUser> signInManager,
+            ILogger<WebAccountManager> logger)
+        {
+            _signInManager = signInManager;
+            _userManager = userManager;
+            _logger = logger;
+        }
+
+        public async Task AddRoleAsync(ApplicationUser user, string role)
+        {
+            await _userManager.AddToRoleAsync(user, role);
```

Comment:
There is a UserManager.AddToRolesAsync we could use instead. Then we pass in the array instead.